specs/agents/tracing-instrumentation-azure.md

```diff
@@ -0,0 +1,420 @@
+# Azure services spans
+
+We describe how to instrument some of Azure's services in this document.
+Some of the services can use existing specs. When there are differences or additions, they have been noted below.
+
+
+----
+
+**NOTE**: 
+Azure Storage can be run as part of [Azure Stack](https://azure.microsoft.com/en-au/overview/azure-stack/). Azure services cannot be inferred from Azure Stack HTTP service endpoints.
+
+----
+
+## Azure Storage
+
+Azure Storage is a collection of storage services, accessed through a
+Storage account. The services include
+
+- Blob storage
+- Queue storage
+- File share storage
+- Table storage
+
+A storage account is created in an Azure location (_region_), but the
+location is not discoverable through REST API calls.
+
+### Blob storage
+
+Object storage for binary and text data via a REST API. There are three types of blobs: block blobs, page blobs and append blobs.
+
+Blobs are organized into containers, where a container can contain an
+unlimited number of blobs, and a storage account can contain an unlimited
+number of containers. Although blob storage is a flat storage scheme
+(only one level of containers), blob names can include path segments (`/`),
+providing a _virtual_ hierarchy.
+
+`<ResourceName>` is the full name of the container and blob. For example, Given container `foo` and blob `bar/baz`, the resource name
+is `foo/bar/baz`.
+
+`<Storage Account Name>` is the leftmost subdomain of the domain of an Azure storage account endpoint.
+
+| APM field | Required? | Format | Notes | Example |
+| --------- | --------- | ------ | ----- | ------- |
+| `span.name` | yes | `AzureBlob <OperationName> <ResourceName>` | Pascal case Operation name | `AzureBlob Upload foo/bar/baz` |

```

Comment:
@felixbarny thoughts on excluding the `<ResourceName>` from the span name? If it were to be removed, is there another appropriate field to capture it?

Reply:
Random thought: The Azure Blob [container](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction#containers) is perhaps *close* to the equivalent of the bucket name that is spec'd to be used in the `span.name` for S3 instrumentation. However, IIUC there could, at least theoretically, be a cardinality issue with that because Azure documents that a "storage account can include an unlimited number of containers", where as, AWS limits to [100 buckets](https://docs.aws.amazon.com/AmazonS3/latest/userguide/BucketRestrictions.html) to start (which can be raised to 1000).

Reply:
container is pretty much Azure's bucket 😄 If AWS spec uses bucket in the span name, it seems reasonable to use container in the span name, where applicable